backend/internal/service/user_service.go

```diff
@@ -735,11 +737,17 @@ func (s *UserService) SignUp(ctx context.Context, signupData dto.SignUpDto, ipAd
 		}
 	}
 
+	displayName := signupData.FirstName
+	if signupData.LastName != nil {
+		displayName = displayName + " " + *signupData.LastName
+	}
+
 	userToCreate := dto.UserCreateDto{
-		Username:  signupData.Username,
-		Email:     signupData.Email,
-		FirstName: signupData.FirstName,
-		LastName:  signupData.LastName,
+		Username:    signupData.Username,
+		Email:       signupData.Email,
+		FirstName:   signupData.FirstName,
+		LastName:    signupData.LastName,
+		DisplayName: displayName,
 	}
 
```

Comment:
Can be reverted since the display name is computed at display time if empty. Better not store extra info in the DB if not needed


```suggestion
	userToCreate := dto.UserCreateDto{
		Username:    signupData.Username,
		Email:       signupData.Email,
		FirstName:   signupData.FirstName,
		LastName:    signupData.LastName,
	}

```